crates/core/tedge_agent/src/tedge_operation_converter/actor.rs

```diff
@@ -560,13 +560,42 @@ impl CommandLog {
         Ok(self.file.as_mut().unwrap())
     }
 
+    async fn log_header(&mut self, topic: &str) {
+        let now = OffsetDateTime::now_utc()
+            .format(&format_description::well_known::Rfc3339)
+            .unwrap();
+        let cmd_id = &self.cmd_id;
+        let operation = &self.operation;
+        let header_message = format!(
+            r#"
+==================================================================
+Triggered {operation} workflow
+==================================================================
+
+topic:     {topic}
+operation: {operation}
+cmd_id:    {cmd_id}
+time:      {now}
+
+==================================================================
+"#
+        );
+        if let Err(err) = self.write(&header_message).await {
+            error!("Fail to log to {}: {err}", self.path)
+        }
+    }
+
     async fn log_state_action(&mut self, state: &GenericCommandState, action: &OperationAction) {
+        if state.is_init() && !self.cmd_id.contains("sub:") {
```

Comment:
I pushed a *fixup* so only `GenericCommandState` is aware of the structure of a command id.